corehq/ex-submodules/casexml/apps/case/util.py

```diff
@@ -281,3 +281,23 @@ def iter_transactions(transactions):
             break
 
     return infos, last_index
+
+
+def get_case_history(case):
+    from casexml.apps.case.xform import extract_case_blocks
+    from corehq.apps.reports.display import xmlns_to_name
+
+    changes = defaultdict(dict)
+    for action in case.actions:
+        case_blocks = extract_case_blocks(action.form)
+        for block in case_blocks:
+            if block.get('@case_id') == case.case_id:
+                property_changes = {}
```

Comment:
probably just the preferred way but for less words/code and quick read, how about
```
property_changes = {
    'Form ID': action.form_id,
    'Form Name': xmlns_to_name(case.domain, action.form.xmlns, action.form.app_id),
#   ....
}
```